docs/platforms/native/advanced-usage/signal-handling/index.mdx

```diff
@@ -0,0 +1,64 @@
+
+* Refrain from registering a conflicting signal handler in your application.
+* If you have to, register your handler after `sentry_init()` and let it invoke Sentry's handler after it finishes, or you won't get a report.
+* If you or one of your dependencies register a handler for the signals listed below _before_ you call `sentry_init()`, we will reinstall that handler as soon as you call `sentry_close()`. Only the `inproc` backend invokes your installed handler at the end of Sentry's handler.
+* If you want to act on any crash and not consider specific signals, [register an `on_crash()` hook](https://docs.sentry.io/platforms/native/configuration/filtering/#using-on_crash) instead of registering a signal handler.
+
+### Signals of Interest
+
+In the context of the Native SDK, we are only interested in signals that will result in the application's unexpected termination (crashes). These typically include:
+
+* `SIGSEGV`: The process tried to access an invalid virtual memory "segment".
+* `SIGBUS`: The process tried to access an invalid physical address or a VM page failed to "page-in".
+* `SIGABRT`: Raised when `abort()` was called. This is the only signal handled on Windows (and only by the `crashpad` backend) because calling `abort()` bypasses Structured Exception Handling ([`SEH`](https://learn.microsoft.com/en-us/windows/win32/debug/structured-exception-handling)).
+* `SIGFPE`: An erroneous arithmetic operation that occurred in the CPU.
+* `SIGILL`: The process attempted to execute an illegal (often privileged) processor instruction.
+* `SIGTRAP`: This is typically used to inform debuggers when execution hits a breakpoint instruction (`int 3`, `brk`, etc.), which instrumentation tools and sanitizers sometimes also issue.
+
+If you use the `crashpad` backend, in addition to the above, on Linux, you will also get crash reports for:
+
+* `SIGSYS`: The kernel received a system call with an invalid argument.
+* `SIGEMT`: An emulator trap occurred.
+* `SIGXCPU`: The process exceeded a previously specified CPU [rlimit](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getrlimit.html).
+* `SIGXFSZ`: Similar to `SIGXCPU`, raised when a process grows a file beyond a maximum allowed size ([`RLIMIT_FSIZE`](https://man7.org/linux/man-pages/man2/getrlimit.2.html)). 
+
+## What to Consider When Writing `on_crash` Hooks
+
+The callback that you register as an `on_crash` hook (and `before_send` if you didn't turn it off for crashes) runs in the context of a signal handler.
+
+Since signal handlers can interrupt the running process at any time (and even each other), take  special care when writing them. 
+
+Our ["develop" docs provide guidelines for writing signal handlers](https://develop.sentry.dev/sdk/signal-handlers/), which also apply to the hooks if you run on Linux or Android. Please take them seriously.
```

Comment:
```suggestion
Our developer documentation provides [guidelines for writing signal handlers](https://develop.sentry.dev/sdk/signal-handlers/). If you run on Linux or Android, these guidelines also apply to hooks. Signal handlers are often difficult to work with, so following these guidelines is important if you want to get good crash reports.
```
Instead of telling them to take it seriously, let's give them a reason to 🙂

Reply:
Okay, let's make it really impactful then: this is not about getting good or better crash reports but rather about causing crashes or other severe faults (like deadlocks) in the handler itself, which can prevent sending reports or programs from terminating.